tests/modules/site-health/audit-autoloaded-options/audit-autoloaded-options-test.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Tests for audit-autoloaded-options module.
+ *
+ * @package performance-lab
+ */
+class Audit_Autoloaded_Options_Tests extends WP_UnitTestCase {
+
+	const WARNING_AUTOLOADED_SIZE_LIMIT_IN_BYTES = 800000;
+
+	/**
+	 * Tests perflab_aao_add_autoloaded_options_test()
+	 */
+	public function test_perflab_aao_add_autoloaded_options_test() {
+		$this->assertEqualSets( Autoloaded_Options_Mock_Responses::return_added_test_info_site_health(), perflab_aao_add_autoloaded_options_test( array() ) );
+	}
+
+	/**
+	 * Tests perflab_aao_autoloaded_options_test() when autoloaded options less than warning size.
+	 */
+	public function test_perflab_aao_autoloaded_options_test_no_warning() {
+		$autoloaded_options_size  = perflab_aao_autoloaded_options_size();
+		$autoloaded_options_count = count( wp_load_alloptions() );
+
+		$this->assertEqualSets(
+			perflab_aao_autoloaded_options_test(),
+			Autoloaded_Options_Mock_Responses::return_perflab_aao_autoloaded_options_test_less_than_limit(
+				$autoloaded_options_size,
+				$autoloaded_options_count
+			)
+		);
+	}
+
+	/**
+	 * Tests perflab_aao_autoloaded_options_test() when autoloaded options more than warning size.
+	 */
+	public function test_perflab_aao_autoloaded_options_test_warning() {
+		self::set_autoloaded_option( self::WARNING_AUTOLOADED_SIZE_LIMIT_IN_BYTES );
+		$autoloaded_options_size  = perflab_aao_autoloaded_options_size();
+		$autoloaded_options_count = count( wp_load_alloptions() );
+
+		$this->assertEqualSets(
+			perflab_aao_autoloaded_options_test(),
+			Autoloaded_Options_Mock_Responses::return_perflab_aao_autoloaded_options_test_bigger_than_limit(
+				$autoloaded_options_size,
+				$autoloaded_options_count
+			)
+		);
+	}
+
+	/**
+	 * Tests perflab_aao_autoloaded_options_size()
+	 */
+	public function test_perflab_aao_autoloaded_options_size() {
+		global $wpdb;
+		$autoloaded_options_size = $wpdb->get_var( 'SELECT SUM(LENGTH(option_value)) FROM ' . $wpdb->prefix . 'options WHERE autoload = \'yes\'' );
+		$this->assertEquals( $autoloaded_options_size, perflab_aao_autoloaded_options_size() );
```

Comment:
To make sure the returned type matches as well.

```suggestion
		$this->assertSame( $autoloaded_options_size, perflab_aao_autoloaded_options_size() );
```

Reply:
fixed